### PR TITLE
Add gamemode handler

### PR DESCRIPTION
**Pull request by joelnir**
Add a gamemode handler for keeping track of existing gamemodes and selecting which one to use.

### CODE REVIEW

File: src/App.test.js [hunk 1]
```diff
@@ -1,9 +1,55 @@
...
+import GamemodeHandler from './game/GamemodeHandler';
+import Game from './game/Game';
 
 it('renders without crashing', () => {
   const div = document.createElement('div');
   ReactDOM.render(<App />, div);
   ReactDOM.unmountComponentAtNode(div);
 });
+
+it('GamemodeHandler is working singleton', () => {
```
Comment (enca11) [inline src/App.test.js]: it('makes sure GamemodeHandler is a working singleton'...

File: src/App.test.js [hunk 2]
```diff
@@ -1,9 +1,55 @@
...
+  const inst1 = GamemodeHandler.getInstance();
+  const inst2 = GamemodeHandler.getInstance();
+
+  assert.equal(inst1, inst2);
+});
+
+it('All gamemodes can be selected', () => {
```
Comment (enca11) [inline src/App.test.js]: it('tests that all gamemodes can be selected'...

File: src/App.test.js [hunk 3]
```diff
@@ -1,9 +1,55 @@
...
+
+  for (let i = 0; i < gmList.length; i += 1) {
+    gmHandler.selectGameMode(gmList[i]);
+
+    // Check so selection is done
+    gmHandler.getSelected();
+  }
+});
+
+it('All gamemodes can be loaded', () => {
```
Comment (enca11) [inline src/App.test.js]: it('tries loading all gamemodes'...

File: src/App.test.js [hunk 4]
```diff
@@ -1,9 +1,57 @@
...
   ReactDOM.render(<App />, div);
   ReactDOM.unmountComponentAtNode(div);
 });
+
+describe('GamemodeHandler', () => {
+  it('is a working singleton', () => {
+    const inst1 = GamemodeHandler.getInstance();
+    const inst2 = GamemodeHandler.getInstance();
+
+    assert.equal(inst1, inst2);
```
Comment (walecome) [inline src/App.test.js]: Suggest using Jests built ins instead of external 'assert' library. https://facebook.github.io/jest/docs/en/expect.html

**Pull request merged**